src/com/sun/jna/NativeLibrary.java

```diff
@@ -898,4 +920,29 @@ else if (Platform.isARM()) {
         
         return cpu + kernel + libc;
     }
+
+    /**
+     * Get the library paths from ldconfig cache. Tested against ldconfig 2.13.
+     */
+    private static ArrayList<String> getLinuxLdPaths() {
+        ArrayList<String> ldPaths = new ArrayList<String>();
+        try {
+                Process process = Runtime.getRuntime().exec("/sbin/ldconfig -p");
+                BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+                String buffer = "";
+                while ((buffer = reader.readLine()) != null) {
+                        int startPath = buffer.indexOf(" => ");
+                        int endPath = buffer.lastIndexOf('/');
+                        if (startPath != -1 && endPath != -1 && startPath < endPath) {
+                                String path =  buffer.substring(startPath+4, endPath);
+                                if (ldPaths.contains(path) == false) {
+                                        ldPaths.add(path);
+                                }
+                        }
+                }
+                reader.close();
+        } catch (Exception e) {
```

Comment:
Are you aware of any execution restrictions on `/sbin/ldconfig`?  SE Linux in particular sometimes has some reduced availability of system-oriented functions.